src/Nethermind/Nethermind.Core.Test/TransactionTests.cs

```diff
@@ -42,4 +43,16 @@ public void Supports1559_returns_expected_results(int decodedFeeCap, bool expect
             Assert.That(transaction.Supports1559, Is.EqualTo(expectedSupports1559));
         }
     }
+
+    public static class TransactionTestExtensions
+    {
+        public static void EqualToTransaction(this Transaction subject, Transaction expectation)
+        {
+            subject.Should().BeEquivalentTo(
+                expectation,
+                o => o.ComparingByMembers<System.Transactions.Transaction>().Excluding((tx) => tx.Data));
+
+            subject.Data.FasterToArray().Should().BeEquivalentTo(expectation.Data.FasterToArray());
```

Comment:
minor, I think you could bring it up to the options to compare Data `WithAutoConversionFor` for example

Reply:
Swapped to `Using` and `WhenTypeIs`.